### ISSUE TITLE
Forced by user

### ISSUE THREAD

**Issue by NaughtyTiger** [VSCodeVim/Vim#5347]
**To Reproduce**

My vscode version is 1.49.3
I enabled setting "editor.formatOnSave" and "auto save"
Then when I double "j" to quit "insert" mode, it doesn't work well. Word "workbench.action.files.save" is shown in statusbar.

**VSCodeVim version**: 1.17.1

<details><summary>Stack trace</summary>

```
Error: Forced by user
	at F.handleKeyEvent (c:\Users\lzhang165060\.vscode\extensions\vscodevim.vim-1.17.1\out\extension.js:9:469893)
	at Object.promise (c:\Users\lzhang165060\.vscode\extensions\vscodevim.vim-1.17.1\out\extension.js:9:235748)
	at runMicrotasks (<anonymous>)
	at processTicksAndRejections (internal/process/task_queues.js:94:5)
	at async Object.runTasks (c:\Users\lzhang165060\.vscode\extensions\vscodevim.vim-1.17.1\out\extension.js:9:621914)
```

</details>

**Comment by NaughtyTiger** [comment]
Stack trace:

  | onDidChangeNotification | @ | workbench.desktop.main.js:sourcemap:3067
-- | -- | -- | --
  | (anonymous) | @ | workbench.desktop.main.js:sourcemap:3066
  | fire | @ | workbench.desktop.main.js:sourcemap:96
  | addNotification | @ | workbench.desktop.main.js:sourcemap:3133
  | notify | @ | workbench.desktop.main.js:sourcemap:4879
  | (anonymous) | @ | workbench.desktop.main.js:sourcemap:4493
  | _showMessage | @ | workbench.desktop.main.js:sourcemap:4492
  | $showMessage | @ | workbench.desktop.main.js:sourcemap:4492
  | _doInvokeHandler | @ | workbench.desktop.main.js:sourcemap:4553
  | _invokeHandler | @ | workbench.desktop.main.js:sourcemap:4553
  | _receiveRequest | @ | workbench.desktop.main.js:sourcemap:4551
  | _receiveOneMessage | @ | workbench.desktop.main.js:sourcemap:4550
  | (anonymous) | @ | workbench.desktop.main.js:sourcemap:4548
  | fire | @ | workbench.desktop.main.js:sourcemap:96
  | fire | @ | workbench.desktop.main.js:sourcemap:452
  | _receiveMessage | @ | workbench.desktop.main.js:sourcemap:457
  | (anonymous) | @ | workbench.desktop.main.js:sourcemap:454
  | fire | @ | workbench.desktop.main.js:sourcemap:96
  | acceptChunk | @ | workbench.desktop.main.js:sourcemap:449
  | (anonymous) | @ | workbench.desktop.main.js:sourcemap:448
  | t | @ | workbench.desktop.main.js:sourcemap:458
  | emit | @ | events.js:223
  | addChunk | @ | _stream_readable.js:309
  | readableAddChunk | @ | _stream_readable.js:290
  | Readable.push | @ | _stream_readable.js:224

**Comment by berknam** [comment]
What is the remap you are using?

**Comment by NaughtyTiger** [comment]
Sorry, I forget I'm using remap. It's like
![image](https://user-images.githubusercontent.com/16698619/95540691-068df880-0a24-11eb-82eb-251411c698e5.png)
It works find until recent. Both commands seems can not be found

**Comment by berknam** [comment]
The recent remapper overhaul allows for infinite recursive remaps, meaning that you can create a remap that calls it self continuously until one of its actions fails or until you cancel it. In normal Vim, in order to cancel you need to press `<C-c>`, in this extension that works too, however there are some users that don't let `<C-c>` be handled by this extension so we needed an alternative and that is `<Esc>`.

When you use that mapping with command `extension.vim_escape` you are not actually remapping `jj` to `<Esc>`, you are mapping it to a list of commands that sends `<Esc>` as if you've pressed it and then saves the files and you are doing it inside a recursive mapping so that `<Esc>` will cancel the recursive mapping. So you have two options, either you change that escape to the 'after' to actually be remapped or you change that remap to be 'NonRecursive'.

TL;DR;
You need to change that remapping to one of this options:
**Option 1:**
```jsonc
"vim.insertModeKeyBindings": [
    {
        "before": [
            "j",
            "j"
        ],
        "after": ["<Esc>"],
        "commands": [
            {
                "command": "workbench.action.files.save"
            }
        ]
    }
],
```
*This option 1 could also be non recursive. In fact if you don't need it to be recursive then you should always use non recursive remaps.*

**Option 2:**
```jsonc
"vim.insertModeKeyBindingsNonRecursive": [
    {
        "before": [
            "j",
            "j"
        ],
        "commands": [
            {
                "command": "extension.vim_escape"
            },
            {
                "command": "workbench.action.files.save"
            }
        ]
    }
],
```

*Sorry for the long explanation, but I think it is important to understand why things behave like this.*

**Comment by NaughtyTiger** [comment]
Thank for your reply. But both way are not working with "auto save" of VSCode
So I just stop save when quitting insert mode.
Anyway, thanks.

**Comment by berknam** [comment]
> 
> 
> Thank for your reply. But both way are not working with "auto save" of VSCode
> So I just stop save when quitting insert mode.
> Anyway, thanks.

I'm sorry, there was a mistake on the settings above. When using the `{ command: string, args: string[] }` you also need to include the 'args' field even if it is just an empty array `"args": []` or you can use just a list of commands like this which looks better:
```jsonc
"vim.insertModeKeyBindings": [
    {
        "before": ["j", "j"],
        "after": ["<Esc>"],
        "commands": ["workbench.action.files.save"]
    }
],
```

Also, if you're using vscode autosave set to `afterDelay` then why do you even need this?

**Comment by NaughtyTiger** [comment]
> Also, if you're using vscode autosave set to `afterDelay` then why do you even need this?
At the beginning, I used remap to autosave. After that I started using autosave of vscode. But remap itself is not harmful, so I forgot to remove autosave in remap. Until recently I encountered a problem with remap exiting insert mode.
